### DISCUSSION TITLE
Trivy should use docker context to select the docker daemon

### DISCUSSION THREAD

**Discussion by PascalTurbo** [aquasecurity/trivy#7685] [Ideas]
### Description

Docker is aware of using multiple docker daemons. These are managed with `docker context` (https://docs.docker.com/engine/manage-resources/contexts/) 
Currently trivy ignores the context and just tries to use the local docker daemon.

For example when using rancher desktop without root permissions, docker socket ist not set at the default path but it's available in users home directory. Currently it's nescessary to get the path manually from the context configuration and pass it to the "--docker-host" option.

### Target

Container Image

### Scanner

None

**Comment by knqyf263** [comment]
Track https://github.com/aquasecurity/trivy/issues/7686
